node/node_test.go

```diff
@@ -167,7 +170,7 @@ func setupFullNodes(t *testing.T, proto protocol.ConsensusVersion, verificationP
 		cfg, err := config.LoadConfigFromDisk(rootDirectory)
 		require.NoError(t, err)
 
-		node, err := MakeFull(logging.Base().With("source", t.Name()+strconv.Itoa(i)), rootDirectory, cfg, []string{}, g)
+		node, err := MakeFull(logging.Base().With("source", t.Name()+strconv.Itoa(i)), rootDirectory, cfg, append(neighbors[:i], neighbors[i+1:]...), g)
```

Comment:
I do not think this works as intended - it keeps copying into the same slice:
```golang
func main() {
	a := []int{1, 2, 3, 4, 5, 6, 7}
	b := append(a[:2], a[3:]...)
	c := append(a[:3], a[4:]...)
	fmt.Printf("%v\n", a)
	fmt.Printf("%v\n", b)
	fmt.Printf("%v\n", c)
}
```
Output
```
[1 2 4 6 7 7 7]
[1 2 4 6 7 7]
[1 2 4 6 7 7]
```

Reply:
Fixed in the most recent patch. Thanks for pointing this out. Let me know if you know of a better way to do this than what I've done which is just `copy` and `append`.